lib/modules/settingsNavigation.js

```diff
@@ -79,96 +120,58 @@ export function makeUrlHash(moduleID?: string, optionKey?: string): string {
 	return hashComponents.join('/');
 }
 
-export function setUrlHash(moduleID?: string, optionKey?: string): void {
-	const titleComponents = ['RES Settings'];
-
-	if (moduleID) {
-		const mod = Modules.getUnchecked(moduleID);
-		const moduleName = mod && i18n(mod.moduleName) || moduleID;
-		titleComponents.push(moduleName);
+export function isSettingsUrl(url: string): boolean {
+	return url.includes('#!settings') /* legacy */ || url.includes(RES_SETTINGS_HASH);
+}
 
-		if (optionKey) {
-			titleComponents.push(optionKey);
+export function updateURL(hash: string) {
+	if (window.top === window) {
+		if (parseHash(location.hash).moduleID === parseHash(hash).moduleID) {
+			history.replaceState(null, '', hash);
+		} else {
+			history.pushState(null, '', hash);
 		}
+	} else {
+		window.parent.postMessage({ hash }, '*');
 	}
-
-	const hash = makeUrlHash(moduleID, optionKey);
-	const title = titleComponents.join(' - ');
-
-	if (location.hash !== hash) {
-		history.pushState(hash, title, hash);
-	}
-}
-
-export function resetUrlHash() {
-	history.pushState('', document.title, location.pathname + location.search);
-}
-
-export function isSettingsUrl(url: string): boolean {
-	return url.includes('#!settings') || url.includes('#res:settings');
 }
 
-function onHashChange() {
-	const hash = location.hash;
-	if (!isSettingsUrl(hash)) return;
-
-	const params = hash.match(/\/(?:\w|\s|%20)+/g);
-	let moduleID;
-	if (params && params[0]) {
-		moduleID = params[0].substring(1).replace('%20', ' ');
+function listener({ origin, data }: MessageEvent) {
+	if (origin !== getOptionsURL().origin) return;
+	const { hash, closing } = (data: any);
+	if (closing) {
+		close();
+	} else {
+		updateURL(hash);
 	}
-	let optionKey;
-	if (params && params[1]) {
-		optionKey = params[1].substring(1).replace('%20', ' ');
-	}
-
-	loadSettingsPage(moduleID, optionKey);
 }
 
-function onPopState(event) {
-	const state = typeof event.state === 'string' && event.state.split('/');
-	if (!state || !isSettingsUrl(state[0])) {
-		if (SettingsConsole.isOpen) {
-			// Avoid adding a duplicate page to the browser history
-			SettingsConsole.close({ resetUrl: false });
-		}
-		return;
+let iframe;
+
+export function open(moduleID?: string, optionKey?: string) {
+	if (iframe || isOptionsPage()) {
+		(iframe && iframe.contentWindow || window).postMessage({ load: { moduleID, optionKey } }, '*');
+	} else {
+		window.addEventListener('message', listener);
+		iframe = document.createElement('iframe');
+		iframe.id = 'console-container';
```

Comment:
@mc10 `#console-container` is the iframe which embeds the options page, but should probably . I have not changed any of class names or ids on the options page, so all ids and class names there stays the same.